horovod/tensorflow/__init__.py

```diff
@@ -121,14 +130,31 @@ def broadcast_global_variables(root_rank):
             root_rank: rank of the process from which global variables will be broadcasted
                        to all other processes.
         """
-        return broadcast_variables(tf.global_variables(), root_rank)
-
-
-if hasattr(tf, 'train') and hasattr(tf.train, 'SessionRunHook'):
-    if hasattr(tf, 'estimator') and hasattr(tf.estimator, 'SessionRunHook'):
-        _SessionRunHook = tf.estimator.SessionRunHook
-    else:
+        if _executing_eagerly():
```

Comment:
Great!  Can you add a test for this functionality (verify that it fails in eager mode)?